lib/collection/src/shard/mod.rs

```diff
@@ -0,0 +1,87 @@
+pub mod local_shard;
+pub mod remote_shard;
+
+use crate::collection_manager::holders::segment_holder::SegmentHolder;
+use crate::shard::remote_shard::RemoteShard;
+use crate::{
+    CollectionInfo, CollectionResult, CollectionSearcher, CollectionUpdateOperations, LocalShard,
+    OptimizersConfigDiff, PointRequest, Record, SearchRequest, UpdateResult,
+};
+use async_trait::async_trait;
+use parking_lot::RwLock;
+use segment::types::{ExtendedPointId, Filter, ScoredPoint, WithPayload, WithPayloadInterface};
+use std::sync::Arc;
+use tokio::runtime::Handle;
+
+pub type ShardId = u32;
+
+pub type PeerId = u32;
+
+/// Shard
+///
+/// A shard can either be local or remote
+///
+#[allow(clippy::large_enum_variant)]
+pub enum Shard {
+    Local(LocalShard),
+    Remote(RemoteShard),
+}
```

Comment:
```suggestion
pub enum Shard {
    Local(Box<LocalShard>),
    Remote(RemoteShard),
}
```

This would help

Reply:
That's a good catch.
I just gave it a try and I feel like adding the `Box` makes things difficult to read in other places.
What about doing it once `RemoteShard` is implemented **if** the warning regarding `large_enum_variant` is still present?

Reply:
I agree